gnd/src/main/java/com/google/android/gnd/ui/offlinearea/OfflineAreaListAdapter.java

```diff
@@ -75,9 +90,42 @@ public void onClick(View v) {
   }
 
   @Override
-  public void onBindViewHolder(ViewHolder viewHolder, int position) {
-    viewHolder.binding.offlineAreaName.setText(offlineAreas.get(position).getId());
-    viewHolder.position = position;
+  public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
+    String areaName =
+        viewHolder.binding.getRoot().getContext().getString(R.string.offline_areas_unknown_area);
```

Comment:
Can you please move this string creation to `catch` block? Otherwise, it seems wasteful to do this init each time.

Reply:
@shobhitagarwal1612 Could you clarify what would be wasted and when?

Reply:
IIUC, this is the default area name "unknown area" which would be set if the try block throws an exception. So, this value would only be used if reverse geocoding fails for a particular point

Reply:
After the refactor in 81c0b3f, the setup is a bit different--we still need to do the allocation early as we can't just hold on to the context in the class (iirc this can lead to leaks)--but it should be a bit cleaner now.